tasty-plutus/src/Test/Tasty/Plutus/Script/Unit.hs

```diff
@@ -164,26 +164,24 @@ shouldn'tValidateTracing ::
   WithScript p ()
 shouldn'tValidateTracing name f td cb = case td of
   SpendingTest {} -> WithSpending $ do
-    tt <- asks (singleTest name . Spender Fail (Just f) td cb)
+    tt <- asks (singleTest name . Spender Bad (Just f) td cb)
     tell . Seq.singleton $ tt
   MintingTest {} -> WithMinting $ do
-    tt <- asks (singleTest name . Minter Fail (Just f) td cb)
+    tt <- asks (singleTest name . Minter Bad (Just f) td cb)
     tell . Seq.singleton $ tt
 
 -- Helpers
 
-data Outcome = Fail | Pass
```

Comment:
`Outcome` just seems identical to `Example` by all means. Is there a good reason to distinguish the "badness" of unit tests and property tests?

Reply:
They are structurally identical, but semantically rather different. `Outcome` is 'what we expect to see from the script running'; `Example` is 'does this particular set of data represent a failing or successful case'. Additionally, `Example` is publically visible, but `Outcome` is only used internally.

I suppose they _could_ be unified, but it would be similar to just using `Bool` for both, which I feel is less clear.

Reply:
My justification is that in unit tests, each test is also a pair of script and test data; so in that sense `Outcome` and `Example` are both describing the "badness" of the pair.

Reply:
Yeah, from that point of view, I can see why you'd want to unify them. This _would_ be a public API break though, so I'll make a separate issue for their unification.